lib/core/src/persist/chain.rs

```diff
@@ -263,6 +263,43 @@ impl Persister {
         Ok(())
     }
 
+    /// Used for Zero-amount Receive Chain swaps, when we fetched the quote and we know how much
+    /// the sender locked up
+    pub(crate) fn update_swap_payer_amount(
+        &self,
+        swap_id: &str,
+        payer_amount_sat: u64,
+    ) -> Result<(), PaymentError> {
+        let swap = self
+            .fetch_chain_swap_by_id(swap_id)?
+            .ok_or_else(|| PaymentError::Generic {
+                err: format!("Cannot update non-existent chain swap with ID: {swap_id}"),
+            })?;
+        ensure_sdk!(
+            matches!(swap.direction, Direction::Incoming),
+            PaymentError::Generic {
+                err: format!(
+                    "Can only update payer_amount_sat for incoming chain swaps. Swap ID: {swap_id}"
+                )
+            }
+        );
```

Comment:
This could be a part of the SQL where clause below

Reply:
Unfortunately if I do that, we won't get an error message if this is called for a swap of the wrong direction.

That's why I chose to explicitly validate.